src/main/java/org/sonatype/nexus/repository/composer/internal/ComposerContentFacetImpl.java

```diff
@@ -136,7 +138,9 @@ protected Content doPutMetadata(final String path,
   {
     StorageTx tx = UnitOfWork.currentTx();
 
-    Asset asset = getOrCreateAsset(path, assetKind);
+    Asset asset = getOrCreateAsset(path);
+
+    asset.formatAttributes().set(P_ASSET_KIND, assetKind.toString());
```

Comment:
Question: is there a reason this has been moved out other than avoiding passing in the assetKind? I think (most?) formats pass in the attribute map to getOrCreateAsset and interate through them adding to the formatAttributes

Reply:
> Question: is there a reason this has been moved out other than avoiding passing in the assetKind? I think (most?) formats pass in the attribute map to getOrCreateAsset and interate through them adding to the formatAttributes

You do have a point. It's been long enough since I wrote this that I'm not totally sure myself, but my suspicion was that I wanted to consolidate all the setting of format attributes in one place, and make that similar across both implementations in the content facet. I don't think that there was a technical reason as such, other than that perhaps I wanted to clear out the existing format attributes on update, so we'd lose the asset kind set in the previous operation if it was a get rather than a create. From there I'm guessing I moved it in both places for the sake of symmetry.

Reply:
> Does this imply that the asset kind can change on a redeploy?

No, that's just where we're setting it (see my other note above). Also note that as the caller is determining the asset kind (which itself comes from being attributed via the route) we should always be passing in the same `AssetKind` for the same path anyway, even on a redeploy scenario.